edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs

```diff
@@ -301,14 +301,27 @@ static string ToProxyLogLevel(LogEventLevel level)
             return proxyLevel;
         }
 
-        V1Volume GetVolume(Mount mount)
+        V1Volume GetVolume(KubernetesModule module, Mount mount)
         {
-            string name = KubeUtils.SanitizeDNSValue(mount.Source);
-            return this.ShouldUsePvc()
-                ? new V1Volume(name, persistentVolumeClaim: new V1PersistentVolumeClaimVolumeSource(name, mount.ReadOnly))
-                : new V1Volume(name, emptyDir: new V1EmptyDirVolumeSource());
-        }
+            string name = KubeUtils.SanitizeK8sValue(mount.Source);
+            string pvcName = KubeUtils.SanitizeK8sValue(module.Name) + name;
 
-        bool ShouldUsePvc() => this.persistentVolumeName.HasValue || this.storageClassName.HasValue;
+            // PVC name will be modulename + volume name
+            // Volume name will be customer defined name or modulename + mount.source
+            if (this.persistentVolumeName.HasValue)
+            {
+                this.persistentVolumeName.ForEach(volumeName => name = volumeName);
+                return new V1Volume(name, persistentVolumeClaim: new V1PersistentVolumeClaimVolumeSource(pvcName, mount.ReadOnly));
+            }
+            else if (this.storageClassName.HasValue)
+            {
+                return new V1Volume(pvcName, persistentVolumeClaim: new V1PersistentVolumeClaimVolumeSource(pvcName, mount.ReadOnly));
```

Comment:
It use to be `name` that was used for volume name, now it is changed to `pvcName`. Could you please explain what is the reason for this change?

Reply:
When storageclass is used, there is no PV. In this case, PVC and PV are having the same name.

Reply:
@CindyXing  Do we need to change logic in the CollectModuleVolumes to use pvcname when querying volumes?

Reply:
@massand, CollectModuleVolumes did take the change in GetVolume.